pass/helpers.go

```diff
@@ -0,0 +1,61 @@
+package pass
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+)
+
+// Used as interface to write gopass secrets
+type secretContainer struct {
+	value string
+}
+
+func (c secretContainer) Bytes() []byte {
+	return []byte(c.value)
+}
+
+// Used as context for secret template
+type templateData struct {
+	Password string
+	Data     map[string]interface{}
+}
+
+// generic function to be used in resource and data source
+func populateResourceData(d *schema.ResourceData, provider *passProvider, path string, readData bool) error {
+	st := provider.store
+	log.Printf("reading %s from gopass", path)
+
+	sec, err := st.Get(context.Background(), path, "") //TODO: support getting a revison via terraform?
+	if err != nil {
+		return errors.Wrapf(err, "failed to read password at %s", path)
+	}
+
+	// Retrieve all data items if keys exist
+	var keys = sec.Keys()
```

Comment:
It looks like `keys` is not used if `readData` is `false`, so you might want to wrap this line in `if readData`